packages/governance/src/governor/extensions/governor_settings.cairo

```diff
@@ -0,0 +1,190 @@
+        impl GovernorSettings: HasComponent<TContractState>,
+        +Drop<TContractState>
+    > of GovernorComponent::GovernorSettingsTrait<TContractState> {
+        /// See `GovernorComponent::GovernorSettingsTrait::voting_delay`.
+        fn voting_delay(self: @GovernorComponentState<TContractState>) -> u64 {
+            let contract = self.get_contract();
+            let this_component = GovernorSettings::get_component(contract);
+
+            this_component.Governor_voting_delay.read()
+        }
+
+        /// See `GovernorComponent::GovernorSettingsTrait::voting_period`.
+        fn voting_period(self: @GovernorComponentState<TContractState>) -> u64 {
+            let contract = self.get_contract();
+            let this_component = GovernorSettings::get_component(contract);
+
+            this_component.Governor_voting_period.read()
+        }
+
+        /// See `GovernorComponent::GovernorSettingsTrait::proposal_threshold`.
+        fn proposal_threshold(self: @GovernorComponentState<TContractState>) -> u256 {
+            let contract = self.get_contract();
+            let this_component = GovernorSettings::get_component(contract);
+
+            this_component.Governor_proposal_threshold.read()
+        }
+    }
+
+    //
+    // External
+    //
+
+    #[embeddable_as(SetSettingsImpl)]
+    impl SetSettings<
+        TContractState,
+        +HasComponent<TContractState>,
+        +GovernorComponent::HasComponent<TContractState>,
+        +GovernorComponent::GovernorCountingTrait<TContractState>,
+        +GovernorComponent::GovernorExecutionTrait<TContractState>,
+        +GovernorComponent::GovernorVotesTrait<TContractState>,
+        +SRC5Component::HasComponent<TContractState>,
+        +Drop<TContractState>
+    > of IGovernorSettingsAdmin<ComponentState<TContractState>> {
+        /// Sets the voting delay.
+        ///
+        /// Emits a `VotingDelayUpdated` event.
+        fn set_voting_delay(ref self: ComponentState<TContractState>, new_voting_delay: u64) {
+            self.assert_only_governance();
+
+            let old_voting_delay = self.Governor_voting_delay.read();
+            self.emit(VotingDelayUpdated { old_voting_delay, new_voting_delay });
+            self.Governor_voting_delay.write(new_voting_delay);
+        }
+
+        /// Sets the voting period.
+        ///
+        /// Requirements:
+        ///
+        /// - `new_voting_period` must be greater than 0.
+        ///
+        /// Emits a `VotingPeriodUpdated` event.
+        fn set_voting_period(ref self: ComponentState<TContractState>, new_voting_period: u64) {
+            self.assert_only_governance();
+            assert(new_voting_period > 0, Errors::INVALID_VOTING_PERIOD);
+
+            let old_voting_period = self.Governor_voting_period.read();
+            self.emit(VotingPeriodUpdated { old_voting_period, new_voting_period });
+            self.Governor_voting_period.write(new_voting_period);
+        }
```

Comment:
Does it make sense to do nothing if `new_voting_period` is equal to `old_voting_period`? Like we do in Vesting and some other places

Same applies also to the `set_voting_delay` and `set_proposal_threshold` functions